lib/muffin_man/request_helpers/outbound_fulfillment/address.rb

```diff
@@ -0,0 +1,73 @@
+module MuffinMan
+  module RequestHelpers
+    module OutboundFulfillment
+      class Address
+        attr_accessor :name, :address_line1, :address_line2, :address_line3, :city,
+                  :district_or_country, :state_or_region, :country_code, :postal_code, :phone
+
+        # Initializes the object
+        # @param [Hash] attributes Model attributes in the form of hash
+        def initialize(attributes = {})
+          return unless attributes.is_a?(Hash)
+
+          if attributes.has_key?('name')
+            self.name = attributes['name']
+          end
```

Comment:
https://developer-docs.amazon.com/sp-api/docs/fulfillment-outbound-api-v2020-07-01-reference#address
Do you think it makes sense to make the required params (name, addressLine1, etc) required here? Either by raising an error if they don't have them, or by just requiring they be passed to initialize the class?

Reply:
I was thinking to have separate validation method to all request objects

like valid? and list_invalid_properties but was not sure, let me know your thoughts on this. @gavinritchie 

# Check to see if the all the properties in the model are valid
+        # @return true if the model is valid
+        def valid?
+          return false if present?(name)
+          return false if address_line1.nil? || address_line1.length == 0
+          return false if state_or_region.nil? || state_or_region.length == 0
+          return false if country_code.nil? || country_code.length == 0
+          true
+        end
+
+        # Show invalid properties with the reasons. Usually used together with valid?
+        # @return Array for valid properties with the reasons
+        def list_invalid_properties
+          invalid_properties = Array.new
+          if name.nil? || name.length == 0
+            invalid_properties.push('invalid value for name')
+          end
+
+          if address_line1.nil? || address_line1.length == 0
+            invalid_properties.push('invalid value for address_line1')
+          end
+
+          if state_or_region.nil? || address_line1.length == 0
+            invalid_properties.push('invalid value for state_or_region')
+          end
+
+          if country_code.nil?
+            invalid_properties.push('invalid value for country_code')
+          end
+
+          invalid_properties
+        end

Reply:
Yeah, actually I like that a lot. Then the end user can choose if they want to validate the address